docs/core/diagnostics/event-counters.md

```diff
@@ -0,0 +1,353 @@
+---
+title: EventCounters in .NET Core
+description: In this article, you'll learn what EventCounters are, how to implement them, and how to consume them.
+ms.date: 07/22/2020
+---
+
+# EventCounters in .NET Core
+
+**This article applies to: ✔️** .NET Core 3.0 SDK and later versions
+
+EventCounters are .NET Core APIs used for lightweight, cross-platform, and near real-time performance metric collection. EventCounters were added as a cross-platform alternative to the "performance counters" of the .NET Framework on Windows. In this article you'll learn what EventCounters are, how to implement them, and how to consume them.
+
+The .NET Core runtime and a few .NET libraries publish basic diagnostics information using EventCounters starting in .NET Core 3.0. Apart from the EventCounters that are provided by the .NET runtime, you may choose to implement your own EventCounters. EventCounters can be used to track various metrics.
+
+EventCounters live as a part of an <xref:System.Diagnostics.Tracing.EventSource>, and are automatically pushed to listener tools on a regular basis. Like all other events on an <xref:System.Diagnostics.Tracing.EventSource>, they can be consumed both in-proc and out-of-proc via <xref:System.Diagnostics.Tracing.EventListener> and EventPipe/ETW (Event Tracing for Windows).
+
+[![EventCounters in-proc and out-of-proc diagram image](media/eventcounters.jpg)](media/eventcounters.jpg#lightbox)
+
+## Runtime counters
+
+The .NET runtime publishes many counters.
+
+### `System.Runtime` providers
+
+- % Time in GC
+- Active Timer Count
+- Allocation Rate
+- Assembly Count
+- CPU usage
+- Exception Rate
+- GC Heap Size
+- Gen 0 GC Rate
+- Gen 0 Heap Size
+- Gen 1 GC Rate
+- Gen 1 Heap Size
+- Gen 2 GC Rate
+- Gen 2 Heap Size
+- LOH Heap Size
+- Monitor Lock Contention Rate
+- ThreadPool Completed Items Rate
+- ThreadPool Queue Length
+- ThreadPool Thread Count
+- Working Set Size
+
+### `Microsoft.AspNetCore.Hosting` providers
+
+The following counters are published as part of [ASP.NET Core](/aspnet/core).
+
+- Current Requests Count
+- Failed Requests Count
+- Requests per second
+- Total Requests Count
+
+### `Microsoft.AspNetCore.Http.Connections` providers
+
+The following counters are published as part of [ASP.NET Core SignalR](/aspnet/core/signalr/introduction).
+
+- Average Connection Duration
+- Total Connections Started
+- Total Connections Stopped
+- Total Connections Timed Out
+
+## EventCounters API overview
```

Comment:
I considered this, however; decided based on the other articles that exist now and the structure they're in - it makes sense to leave this as is.

Reply:
Is that because you think this arrangement is the better flow or because it would be overly laborious to modify it or something else? I'm not pushing back but understanding the rationale helps me know whether if we'll want to revisit this at some point in the future. Thanks!